### PR TITLE
Fix SqlGuid byte order mismatch across the SQL_C_GUID ODBC boundary

### LINKED ISSUE #596: SqlGuid: TryParse/format() and Create()/ODBC binding disagree on byte order, corrupting round-tripped GUIDs
## Summary

`SqlGuid::Create()` (the Win32 native-GUID path) and `SqlGuid::TryParse()` / the `std::formatter<SqlGuid>` specialization write and read `data[16]` in two different, incompatible byte orders. Any GUID that is formatted to text via `to_string()`/`std::format` and later re-parsed via `TryParse()` (or vice versa) round-trips to a *different* physical GUID once it crosses the ODBC `SQL_C_GUID` boundary, even though the two textual representations look unrelated at a glance.

## Root cause

- `Create()`'s Win32 path (`src/Lightweight/DataBinder/SqlGuid.cpp`) builds `data[]` by splitting a `GUID`'s `Data1`/`Data2`/`Data3` fields **big-endian**:
  ```cpp
  guid.data[0] = winGuid.Data1 >> 24;
  guid.data[1] = winGuid.Data1 >> 16;
  guid.data[2] = winGuid.Data1 >> 8;
  guid.data[3] = winGuid.Data1;
  ```
  This matches the native `GUID`/`SQL_C_GUID` in-memory layout, so `Create()`'s output binds correctly via `InputParameter`'s `SQLBindParameter(..., SQL_C_GUID, ...)`.

- `TryParse()` (`SqlGuid.hpp`/`SqlGuid.cpp`) instead fills `data[]` straight from consecutive hex-pair positions in the **text**, i.e. `data[0]` is the first byte as written in the string (`xxxxxxxx-...`) — the opposite convention from `Create()`.

- `std::formatter<SqlGuid>::format()` (`SqlGuid.hpp`) reconstructs `Data1`/`Data2`/`Data3` from `data[]` by reversing the byte order:
  ```cpp
  (uint32_t) guid.data[3] | (uint32_t) guid.data[2] << 8 |
      (uint32_t) guid.data[1] << 16 | (uint32_t) guid.data[0] << 24
  ```
  This is the correct inverse of `Create()`'s layout, but the *wrong* inverse of `TryParse()`'s layout.

So `TryParse` and `format`/`to_string` form a self-consistent round-trip pair (parse text -> format back to the same text), and `Create` and the ODBC binding form a separate self-consistent pair (generate -> bind -> matches what SQL Server itself considers that GUID to be) - but the two pairs disagree with each other. A GUID created via `Create()`, inserted, then later formatted via `to_string()` for display/storage-as-text (e.g. embedding a `uniqueidentifier` primary key in an unrelated text/XML field) produces a string that does **not** match what `SELECT ... WHERE col = '<that string>'` finds in the database - because `format()` un-reverses bytes that `Create()` never reversed in the first place.

## Reproduction

Observed downstream in an application: a `SqlGuid::Create()`'d value was inserted as a `uniqueidentifier` primary key, then `Light::to_string()`'d for storage elsewhere. Querying the target table with the exact string `to_string()` produced returned 0 rows; the actual stored row's `NR`, as SQL Server itself renders it (`sqlcmd`, `SELECT NR FROM t`), was:

```
to_string() output:  4B0BA16D-1A21-4BDA-881D-5550E28B7858
SQL Server's own value: 6DA10B4B-211A-DA4B-881D-5550E28B7858
```

Confirmed via direct query: `SELECT NR FROM t WHERE NR = '6DA10B4B-211A-DA4B-881D-5550E28B7858'` matches the row; `... WHERE NR = '4B0BA16D-1A21-4BDA-881D-5550E28B7858'` returns nothing. Note the first three hex groups are exactly byte-reversed between the two strings (`4B0BA16D` <-> `6DA10B4B` byte-swapped, etc.); the last two groups (`881D-5550E28B7858`) are identical, matching `format()`'s big-endian-only treatment of `data[10..15]`.

## Why this matters beyond cosmetics

- **Round-trips correctly only through one path at a time.** An app that always uses `TryParse`+`format`/`to_string` together (never touching `Create()`/ODBC binding for the same value) or always uses `Create()`+ODBC binding together (never formatting to text) won't notice. Mixing them - e.g. `Create()` a GUID, insert it, then `to_string()` it for embedding in a different record as a foreign reference, then `TryParse()` that string back later to query - silently produces a lookup that either fails or (worse, if both write and read paths apply the *same* internal inconsistency) succeeds while the *portable, cross-tool-correct* string is wrong.
- **Breaks cross-backend/cross-tool correctness.** SQL Server's `uniqueidentifier` mixed-endian convention (`Data1`/`Data2`/`Data3` little-endian in storage, last two groups big-endian) is the reason this asymmetry exists at all; it doesn't apply to Postgres `uuid` or SQLite (stored as plain text via `to_string()` per this file's own SQLITE branches). Any consumer outside this library that reads the raw column with a standards-correct driver, or the same value round-tripped through a different backend, will see a different/wrong GUID.

## Suggested fix

Make `TryParse()` fill `data[]` in the same physical byte order `Create()` uses (i.e. apply the same big-endian folding for the first three groups that `Create()`'s Win32 path already does), so `Create()`, `TryParse()`, `format()`/`to_string()`, and the ODBC `SQL_C_GUID` binding all agree on one internal representation. Add a round-trip test: `TryParse(x)` -> ODBC insert -> read back via a raw/independent query (or at minimum assert `to_string(TryParse(x).value()) == x` *and* that the resulting bytes match what `Create()` would produce for an equivalent value) to catch this class of regression.

## Environment

- Pinned vcpkg tag in the downstream project: `v0.20260625.0`
- Confirmed still present on `Lightweight`'s current `master` (`8b76d5ca`) as of this report.

### PR DESCRIPTION
**Pull request by Yaraslaut**
`SqlGuid::data` is stored in canonical (textual/RFC4122) byte order — the order `TryParse()`, `UnsafeParse()`, and the `std::formatter` specialization all agree on. The ODBC `SQL_C_GUID` C type, however, is a native `SQLGUID` struct whose `Data1`/`Data2`/`Data3` fields are transmitted in native (little-endian) byte order, which is the byte-reverse of the textual order for those three groups. Every place that bound or fetched raw `SQL_C_GUID` bytes passed `SqlGuid::data` straight through with no conversion, so a GUID generated via `Create()`/`TryParse()` and later formatted back to text no longer matched what MSSQL or PostgreSQL themselves considered that GUID to be, once round-tripped through the database — exactly as reported in #596.

## Changes

- Add `detail::SwapGuidWireByteOrder()` (self-inverse) in `SqlGuid.hpp`.
- Apply it at every raw `SQL_C_GUID` crossing point: `SqlDataBinder<SqlGuid>::InputParameter/OutputColumn/GetColumn`, `RowArrayCursor::GetGuid`'s array-fetch path, and the backup restore path's `GuidBatchColumn::ToRaw()`.
- Fix an adjacent gap where `OutputColumn` had no `SqlServerType::UNKNOWN` case (unlike its `InputParameter`/`GetColumn` siblings), which fell through to `std::unreachable()`.
- Add unit tests for the byte-swap helper and a DB-level regression test that compares an inserted `SqlGuid` against a literal-string `WHERE` clause — independent of the library's own binder — which reproduces the issue's exact byte-reversal pattern on both MSSQL and PostgreSQL before this fix.

Fixes #596

### CODE REVIEW

**Review by christianparpart — APPROVED** [review]

File: src/Lightweight/DataBinder/SqlGuid.hpp [hunk 1]
```diff
@@ -137,6 +139,30 @@ inline LIGHTWEIGHT_FORCE_INLINE std::string to_string(SqlGuid const& guid)
...
+    /// apply this exactly once when crossing that boundary, in either direction.
+    ///
+    /// @param data The 16-byte GUID payload to convert in place.
+    inline void SwapGuidWireByteOrder(std::span<uint8_t, 16> data) noexcept
+    {
+        // Data1 (4 bytes), Data2 (2) and Data3 (2) differ between the two representations;
+        // Data4 (the trailing 8) is a plain byte array in both and is left untouched.
+        std::ranges::reverse(data.first<4>());
+        std::ranges::reverse(data.subspan<4, 2>());
+        std::ranges::reverse(data.subspan<6, 2>());
```
[the remark this answers is not in the document]
Comment (Yaraslaut) [reply src/Lightweight/DataBinder/SqlGuid.hpp:169]: Good catch — fixed in 002e2829: `SwapGuidWireByteOrder` now has a `static_assert(std::endian::native == std::endian::little)` making the assumption explicit and compile-time enforced, matching the existing little-endian-only precedent in `SqlNumeric.hpp`. Going fully portable (branching the swap on `std::endian::native`) is possible but out of scope here since nothing in this project currently targets a big-endian host.
